### PR TITLE
Api 3574 service history controller try two

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
# Description

Implementation of service_history controller.

## Quick Start

- run mock-mpi
- run mock-emis
- `export VETERAN_VERIFICATION_HOST=http://localhost:8080`

```sh
curl -s -H 'Accept: application/json' $VETERAN_VERIFICATION_HOST/v0/service_history/1012832025V743496 | jq . 
```

### CODE REVIEW

File: veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/ServiceHistoryIT.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        "COMPLETION OF REQUIRED ACTIVE SERVICE",
+        serviceHistory.data().get(1).attributes().separationReason());
+    assertEquals(0, serviceHistory.data().get(1).attributes().deployments().size());
+    assertEquals("fcc93b2c-9aa8-5d7b-ab26-8e5577b660b8", serviceHistory.data().get(2).id());
+    assertEquals("service-history-episodes", serviceHistory.data().get(2).type());
+    assertEquals("Sam", serviceHistory.data().get(2).attributes().firstName());
+    assertEquals("Gardner", serviceHistory.data().get(2).attributes().lastName());
+    assertEquals(
+        "Army National Guard", serviceHistory.data().get(2).attributes().branchOfService());
+    assertEquals("2007-03-15", serviceHistory.data().get(2).attributes().startDate().toString());
```
Comment (dbrown-va) [inline veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/ServiceHistoryIT.java:153]: One difference between vets-api today and the Java Service History.

vets-api returns a null end_date, where as the java implementation is omitting the field from the response.
```
"end_date": null,
```

The transformer seems to be populating a null there by default, and the schema has it required but nullable.

Looking to see if there is a setting to include null fields in the json response.

File: veteran-verification-api/src/main/java/gov/va/api/lighthouse/veteranverification/api/ApiError.java [hunk 2]
```diff
@@ -64,6 +64,24 @@ public InvalidParameterApiError(String value, String field) {
     }
   }
 
+  /** Error class for an no service history found. */
```
Comment (beaugrantham-va) [inline veteran-verification-api/src/main/java/gov/va/api/lighthouse/veteranverification/api/ApiError.java:67]: > for an no

Comment (dbrown-va) [reply veteran-verification-api/src/main/java/gov/va/api/lighthouse/veteranverification/api/ApiError.java]: Removed `an`

File: veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/ServiceHistoryIT.java (hunk 1 above)
Comment (dbrown-va) [reply veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/ServiceHistoryIT.java:153]: Adding a `@JsonInclude` annotation to the endDate in the ServiceHistoryResponse class can enforce it being included, even if null.

However, I thought including nulls was the default behavior.

And only when `@JsonInclude(value= JsonInclude.Include.NON_EMPTY)` would make it not include.

Comment (dbrown-va) [reply veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/ServiceHistoryIT.java:153]: Confirmed the exclude nulls comes from the parent pom.
https://github.com/department-of-veterans-affairs/health-apis-parent/blob/master/service-auto-config/src/main/java/gov/va/api/health/autoconfig/configuration/JacksonConfig.java#L96

We want to add that `@JsonInclude` annotation to match vets-api.

**Review by dbrown-va — CHANGES_REQUESTED** [review]
One change requested for the `@JsonInclude` addition.

**Review by dbrown-va — APPROVED** [review]

**Pull request merged**